internal/codespaces/connection/connection.go

```diff
@@ -74,6 +79,38 @@ func NewCodespaceConnection(ctx context.Context, codespace *api.Codespace, httpC
 	}, nil
 }
 
+// Connect connects the client to the tunnel.
+func (c *CodespaceConnection) Connect(ctx context.Context) error {
```

Comment:
What's protecting `c.TunnelClient.connected` racing with `c.TunnelClient.connected = true` when this function is called from inside go routine in `UpdatePortVisibility`?

https://github.com/cli/cli/pull/8275/files#diff-010c36da4b1a993e4a4a6b18584ed40b57a2f813bb8c231ec32d85ec624492dcL260

Reply:
Agreed, it seems like this should be locked with a mutex to prevent concurrent connection attempts.

Reply:
Good idea, updated to add a mutex around the connection logic